### ISSUE TITLE
InkWell is not announced as a button on iOS even with a tap handler

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#45823]
Internal: b/145406913

I just tried this on a hello_flutter app. The following widget:

```dart
InkWell(
  child: Text('some text'),
  onTap: () {},
)
```
results in "some text", "double tap to activate" on Android Talkback and "some text" on iOS.

It seems like on Android there's a "tappable" boolean but on iOS there isn't?
If we need to announce this as a button on iOS, should we do the same on Android?

**Comment by mehmetf** [comment]
Customer:Money just said:

> I remember testing this in our original investigation, and I'm quite sure InkWells *used* to say 'Button' on iOS back in June/July timeframe. Note that the Semantic Debugger will show 'some text (Button)' for the code example above.

Marking this as regression.

**Comment by jkurtw** [comment]
This issues is the cause of 9 'launch blocking' issues for `customer: money (g3)`

**Comment by goderbauer** [comment]
InkWell itself doesn't identify as a button because not everything that may respond with an InkWell upon touch is a button (e.g. a Checkbox). If you want an InkWell to identify itself as a button you'll have to wrap it in `Semantics(button: true, child: InkWell(...))`.

The SemanticsDebugger does have a bug, though: It incorrectly assumes that everything with a tap handler is a button. I filed https://github.com/flutter/flutter/issues/46039 for that.

I also don't think this is a regression. From what I can tell by going through the git history this hasn't changed in the last year.

I am closing this as working as intended. Feel free to re-open with more details if I miss-understood the issue.

**Comment by lukepighetti** [comment]
We use InkWell liberally as a way to build tappable functionality. I thought this had full support for button semantics, but I see now that it does not. What are we missing when we use InkWell for this purpose?
